test/jdk/java/util/Collection/MOAT.java

```diff
@@ -887,6 +887,41 @@ private static void testStringElement(Collection<Integer> c) {
         catch (Throwable t) { unexpected(t); }
     }
 
+    private static void testAddAll(Collection<Integer> c) {
+        if (!supportsAdd(c))
+            return;
+
+        clear(c);
+
+        // Test empty ArrayList source
+        ArrayList<Integer> emptySource = new ArrayList<>();
+        check(!c.addAll(emptySource));
```

Comment:
It seems like there's an assertion missing here. We check the return value from addAll(), but not the state of the collection after the operation. Maybe something like `check(c.isEmpty())` ?